### PR TITLE
feat: add utils for getting the intermediary UO struct

### PR DESCRIPTION
**Pull request by moldy530**
## PR-Codex overview
This PR focuses on adding new types and utility functions to the core package. 

### Detailed summary
- Added new types and utility functions in `index.ts` and `types.ts` files.
- Updated the `AccountMiddlewareFn` and `AccountMiddlewareOverrideFn` types in `provider/types.ts`.
- Updated the `SmartAccountProvider` class in `provider/base.ts` to include new methods and changes related to user operations.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by moldy530** [comment]
Current dependencies on/for this PR:
* main
  * **PR #86** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/86" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/86?utm_source=stack-comment).

File: packages/core/src/provider/base.ts [hunk 1]
```diff
@@ -352,15 +379,13 @@ export class SmartAccountProvider<
   // You should implement your own middleware to override these
   // or extend this class and provider your own implemenation
   readonly dummyPaymasterDataMiddleware: AccountMiddlewareFn = async (
-    struct: UserOperationStruct
-  ): Promise<UserOperationStruct> => {
+    struct
```
Comment (avasisht23) [inline packages/core/src/provider/base.ts:382]: nit: we don't want to type the inputs?

File: packages/core/src/provider/base.ts [hunk 2]
```diff
@@ -352,15 +379,13 @@ export class SmartAccountProvider<
...
+  ) => {
     struct.paymasterAndData = "0x";
     return struct;
   };
 
-  readonly paymasterDataMiddleware: AccountMiddlewareFn = async (
-    struct: UserOperationStruct
-  ): Promise<UserOperationStruct> => {
+  readonly paymasterDataMiddleware: AccountMiddlewareFn = async (struct) => {
```
Comment (avasisht23) [inline packages/core/src/provider/base.ts:388]: nit: we don't want to type the inputs?

**Review by avasisht23 — APPROVED** [review]

File: packages/core/src/provider/base.ts (hunk 1 above)
Comment (moldy530) [reply packages/core/src/provider/base.ts:382]: it's already typed because of the interface that this class implements

**Pull request merged**